### PR TITLE
Improve fuse mount based on user configuration

### PR DESCRIPTION
**Pull request by LuQQiu**
### What changes are proposed in this pull request?

User input > user alluxio config > alluxio default config when setting the Fuse mount options
alluxio.fuse.mount.point
alluxio.fuse.mount.alluxio.path
alluxio.fuse.mount.options

### Why are the changes needed?

To provide a central way of configuring worker fuse and standalone fuse.

### Does this PR introduce any user facing changes?

Yeah, change how users launch Kubernetes scripts

### CODE REVIEW

**Comment by LuQQiu** [comment]
FYI @liugs0213 @maobaolong @Nizifan this PR may break the kubernetes backward compatibility.
Instead of using environment variable to set the alluxio path and fuse mount point, we took the following configuration
```
alluxio.fuse.mount.point
alluxio.fuse.mount.alluxio.path
alluxio.fuse.mount.options
```
This is for code cleanness and for better maintainability

File: integration/kubernetes/helm-chart/alluxio/templates/fuse/daemonset.yaml [hunk 1]
```diff
@@ -109,10 +109,6 @@ spec:
             valueFrom:
               fieldRef:
                 fieldPath: status.hostIP
-          - name: FUSE_ALLUXIO_PATH
-            value: {{ .Values.fuse.alluxioPath }}
-          - name: MOUNT_POINT
-            value: {{ .Values.fuse.mountPath }}
```
Comment (LuQQiu) [inline integration/kubernetes/helm-chart/alluxio/templates/fuse/daemonset.yaml:115]: @liugs0213 @maobaolong @Nizifan those environment variables will not set the alluxioPath, mountPoint, requires alluxio.fuse.mount.point and alluxio.fuse.mount.alluxio.path

File: integration/fuse/bin/alluxio-fuse [hunk 2]
```diff
@@ -19,13 +19,13 @@ umount \tunmount an Alluxio path from local file system
 stat \tshow status of Alluxio mount points"
 
 readonly MOUNT_USAGE="
-Mounts a path in Alluxio namespace (defaults to "/") to mount point on local file system.
-When no argument is given, list the current mount point
```
Comment (LuQQiu) [inline integration/fuse/bin/alluxio-fuse:23]: @liugs0213 @maobaolong @Nizifan
when no argument is given, will not list the current mount point, but use the default `alluxio.fuse.mount.point` and `alluxio.fuse.mount.alluxio.path` to mount.

File: integration/kubernetes/helm-chart/alluxio/templates/fuse/daemonset.yaml [hunk 3]
```diff
@@ -125,7 +121,7 @@ spec:
           lifecycle:
             preStop:
               exec:
-                command: ["/opt/alluxio/integration/fuse/bin/alluxio-fuse", "unmount", "{{ .Values.fuse.mountPath }}"]
+                command: ["/opt/alluxio/integration/fuse/bin/alluxio-fuse", "unmount", "-f", "{{ .Values.fuse.mountPath }}"]
```
Comment (LuQQiu) [inline integration/kubernetes/helm-chart/alluxio/templates/fuse/daemonset.yaml:124]: As a temporary fix, better to be soft umount first and then hard umount

**Review by ssz1997 — CHANGES_REQUESTED** [review]
Well something is broken

File: integration/fuse/bin/alluxio-fuse [hunk 4]
```diff
@@ -54,36 +54,48 @@ check_fuse_jar() {
...
+  local mount_point=""
+  local mount_alluxio_path=""
+  if [[ $# -ge 2 ]]; then
+    mount_point="$1"
+    mount_alluxio_path="$2"
+  elif [[ $# -eq 1 ]]; then
+    mount_point="$1"
   else
-    full_mount_option="big_writes"
+    mount_point=$(${BIN}/alluxio getConf alluxio.fuse.mount.point)
```
[the remark this answers is not in the document]
Comment (LuQQiu) [reply integration/fuse/bin/alluxio-fuse:69]: getConf will process the environment variable and default alluxio configuration, why this command cannot get the mount point that user set?
if alluxio.fuse.mount.point is configured in ALLUXIO_FUSE_JAVA_OPTS, can we run this command to get the value?

**Review by ssz1997 — APPROVED** [review]
LGTM. Thanks for the improvement, mount and code

**Comment by LuQQiu** [comment]
Error: 5.298 [ERROR] Errors: 
Error: 5.298 [ERROR] alluxio.client.cli.job.DistributedCommandsStatsTest.testAsyncPersistCancelStats
Error: 5.301 [ERROR]   Run 1: DistributedCommandsStatsTest.testAsyncPersistCancelStats:221 » Timeout Timed o...
Error: 5.303 [ERROR]   Run 2: DistributedCommandsStatsTest.testAsyncPersistCancelStats:221 » Timeout Timed o...

**Comment by LuQQiu** [comment]
alluxio-bot, merge this please

File: integration/kubernetes/helm-chart/alluxio/templates/fuse/daemonset.yaml (hunk 1 above)
Comment (LuQQiu) [reply integration/kubernetes/helm-chart/alluxio/templates/fuse/daemonset.yaml:115]: Changes will be reverted, those parameters will still be here to support multiple Fuse yaml mounts/files with one alluxio-configmap.yaml

**Pull request merged**